libpod/container_internal_linux.go

```diff
@@ -39,7 +39,10 @@ func (c *Container) cleanupCgroups() error {
 	}
 
 	if err := cgroup.Delete(); err != nil {
-		return err
+		// For now we are going to only warn on failures to clean up cgroups
+		// We have a conflict with running podman containers cleanup in same cgroup as container
+		logrus.Warnf("Ignoring Error cleaning up container %s CGroups: %v", c.ID(), err)
+		return nil
```

Comment:
You changed the behaviour. Now we do not cleanup storage if the cgroup cleanup fails.  Why did you make this change?

It might be better to go back to the original code.  SInce I am not totally sure if this ignore is necessary since we removed cleanup from stop.  I did not have time to check it out.

You could test the theory by running a container in foreground and -d daemon mode, then stop both, if this log message does not show up, then we should go back to failing on error.  But either way it should not be in the PR,

Reply:
> You changed the behaviour.

Did I?  In master, this error [causes `cleanupCgroups` to error][1], which [we then ignore][2].  Now we ignore  this error and return early here.  I think those are the same.

My new code is making *other* errors from this helper fatal again (e.g. [this one][3]), but I didn't think #942 cared whether they were fatal or not.  It only spoke about failed removal.

> Now we do not cleanup storage if the cgroup cleanup fails.

I think we will for `cgroup.Delete` errors.

> Why did you make this change?

[Because][4] ignoring cgroup-deletion errors is, like the rest of our cgroup logic, Linux-specific.

> You could test the theory by...

I don't have a working `podman` locally, but I'll see what I can find in the CI logs.

[1]: https://github.com/projectatomic/libpod/blob/b96be3af1b9d00662758211420c955becbaf2f9e/libpod/container_internal_linux.go#L42
[2]: https://github.com/projectatomic/libpod/blob/b96be3af1b9d00662758211420c955becbaf2f9e/libpod/container_internal.go#L899
[3]: https://github.com/projectatomic/libpod/blob/b96be3af1b9d00662758211420c955becbaf2f9e/libpod/container_internal_linux.go#L38
[4]: https://github.com/projectatomic/libpod/pull/1032#issuecomment-401479031